zimbra/src/main/java/fr/openent/zimbra/tasks/service/impl/RecallMailServiceImpl.java

```diff
@@ -123,8 +134,11 @@ public Future<List<RecallMail>> getRecallMailsForOneStructure (String structureI
 
     private Future<Message> fetchUserIdForRecalledMessage(Message message) {
         Promise<Message> promise = Promise.promise();
-
         recipientService.getUserIdsFromEmails(message.getAllAddresses())
+                .compose(userMap -> {
+                    String senderMail = message.getEmailAddresses().stream().filter(mail -> mail.getAddrType().equals(ADDR_TYPE_FROM)).findFirst().get().getAddress();
```

Comment:
get() peut potentiellement être null 

Il vaut mieux faire une gestion de vérif comme le recommande l'objet `Optional`
```suggestion
       String senderMail = message.getEmailAddresses().stream()
                            .filter(mail -> mail.getAddrType().equals(ADDR_TYPE_FROM))
                            .findFirst()
                            .orElse(ZimbraEmail.fromZimbra(new JsonObject()))
                            .getAddress();
```

on remplace `get` par `orElse` pour avoir une valeur par (défaut) dans le cas où tu tenterais de faire un appel à `getAddress()` donc potentiel NPE